### PR TITLE
fix: don't wrap licenses in parentheses (#10)

### LINKED ISSUE #10: Don't Wrap Licenses in Parentheses When Parsing Arrays
**Describe the bug**

When the license (or, more likely, the deprecated licenses property) of a manifest contains an array as opposed to a string, it is currently wrapped it in parentheses.

**To reproduce**

1. Run `yarn plugin import https://raw.githubusercontent.com/tophat/yarn-plugin-licenses/5f0ce9f190984299a4b200bdd87765d1d1f30060/bundles/@yarnpkg/plugin-licenses-audit.js` to import this plugin.
2. Run `yarn add node-exit@0.1.2` to install the [node-exit]((https://github.com/cowboy/node-exit/blob/v0.1.2/package.json)) package at v0.1.2.
3. Run `yarn licenses audit --summary` to run this plugin.

**Expected behavior**

The second column contains "MIT" rather than "(MIT)" for node-exit@0.1.2 as listed in the type field of its package.json. A case such as this one where the licenses array is of length 1 yields awkward output that may complicate the writing of config files for downstream developers.

**Additional context**

When I accidentally introduced this issue in a743c2edf99d116f9114695072a4a7be4765a123, I was under the mistaken impression that the parentheses around SPDX expressions were required based on [npm's documentation of package.json](https://docs.npmjs.com/cli/v7/configuring-npm/package-json#license). [The SPDX specification](https://spdx.github.io/spdx-spec/appendix-IV-SPDX-license-expressions/) says outer parentheses are optional. exit@0.1.2 is an indirect dependency of [jest@27.0.0-next.8](https://github.com/facebook/jest/tree/v27.0.0-next.8).

### PR DESCRIPTION
**Pull request by Kurt-von-Laven**
## Description

If the license (or, more likely, the deprecated licenses property) of a manifest contains an array, then don't wrap the SPDX expression we emit in unnecessary parentheses. This is most awkward when the array is of length 1. For example, while "(MIT)" is technically a valid SPDX expression, it is a counterintuitive way of expressing "MIT" and may complicate the implementation of some downstream config files.

## Related Issues

- Closes #10, don't wrap licenses in parentheses when parsing arrays.

## Checklist

- [x] I agree to abide by the [Code of Conduct](https://github.com/tophat/yarn-plugin-licenses/blob/master/CODE_OF_CONDUCT.md).
- [x] This PR has sufficient documentation.
- [x] This PR has sufficient test coverage.
- [x] This PR title satisfies semantic [convention](https://www.conventionalcommits.org/en/v1.0.0/#summary).

## Additional Comments

An alternative solution would be to only remove the counterintuitive parentheses in the case of arrays of length 1, and for longer arrays to prefer the style used in [npm's package.json documentation](https://docs.npmjs.com/cli/v7/configuring-npm/package-json#license) over emitting the simplest valid SPDX expression according to [the spec](https://spdx.github.io/spdx-spec/appendix-IV-SPDX-license-expressions/). I have not surveyed the frequency with which package authors adhere to one style over the other.

You must have fixed the permission issue I ran into when attempting to `yarn install` so I could test, lint, and build my previous submission. I did have to copy the pertinent scripts out of the package.json and precede them by `yarn run` in case that helps anyone in the future.

### CODE REVIEW

**Comment by Kurt-von-Laven** [comment]
What is your opinion on this pull request, @noahnu?

**Comment by noahnu** [comment]
> An alternative solution would be to only remove the counterintuitive parentheses in the case of arrays of length 1

I prefer that if you want to make that change to the PR. I.e. omit parentheses only if the array has length 1.

> for longer arrays to prefer the style used in npm's package.json documentation 

The style for multiple licenses is the one with brackets that we currently have. The array format is deprecated.

**Comment by noahnu** [comment]
> I did have to copy the pertinent scripts out of the package.json and precede them by yarn run in case that helps anyone in the future.

@Kurt-von-Laven What's this permission issue?

**Comment by Kurt-von-Laven** [comment]
> > An alternative solution would be to only remove the counterintuitive parentheses in the case of arrays of length 1
> 
> I prefer that if you want to make that change to the PR. I.e. omit parentheses only if the array has length 1.
> 

Done!

> > for longer arrays to prefer the style used in npm's package.json documentation
> 
> The style for multiple licenses is the one with brackets that we currently have. The array format is deprecated.

I was referring to the output rather than the input style of the license from the perspective of this plugin. npm's package.json documentation gives `"(ISC OR GPL-3.0)"` as an example, but the simplest valid SPDX expression is "ISC OR GPL-3.0".

> > I did have to copy the pertinent scripts out of the package.json and precede them by yarn run in case that helps anyone in the future.
> 
> @Kurt-von-Laven What's this permission issue?

```
$ yarn run build
permission denied: builder
$ yarn run builder build plugin
➤ YN0000: ┌ Building @yarnpkg/plugin-licenses-audit
➤ YN0000: │ webpack: setup
➤ YN0000: │ DeprecationWarning: optimizeChunkAssets is deprecated (use Compilation.hooks.processAssets instead and use one of Compilation.PROCESS_ASSETS_STAGE_* as stage option)
➤ YN0000: │ webpack: building
➤ YN0000: │ DeprecationWarning: Module.errors was removed (use getErrors instead)
➤ YN0000: │ webpack: sealing
➤ YN0000: │ webpack: emitting
➤ YN0000: │ webpack: done
➤ YN0000: │ webpack: cache
➤ YN0000: └ Completed in 4s 880ms

➤ YN0000: ✓ Done building @yarnpkg/plugin-licenses-audit!
➤ YN0000: ? Bundle path: /home/kurt/Documents/tophat/yarn-plugin-licenses/bundles/@yarnpkg/plugin-licenses-audit.js
➤ YN0000: ? Bundle size: 142.24 KB
$ yarn run test
permission denied: jest
$ yarn run jest
 PASS  tests/plugin.test.ts
  Yarn Licenses Audit Plugin
    ✓ exports a plugin (1 ms)

Test Suites: 1 passed, 1 total
Tests:       1 passed, 1 total
Snapshots:   0 total
Time:        2.145 s
Ran all test suites.
```

We encountered the same thing in our own project, and addressed it by starting most of our scripts with `yarn run`, but I could definitely believe this is an issue with my own development environment.

File: src/index.ts [hunk 1]
```diff
@@ -308,7 +308,12 @@ class AuditLicensesCommand extends Command<CommandContext> {
                 }
                 return licenseTypes
             }, [])
-            return licenseTypes.length ? `(${licenseTypes.join(' AND ')})` : ''
+            if (!licenseTypes) {
```
Comment (noahnu) [inline src/index.ts:311]:
```suggestion
            if (!licenseTypes.length) {
```

**Pull request merged**